api/test/wfl/tools/fixtures.clj

```diff
@@ -193,6 +193,10 @@
    datarepo/delete-snapshots-then-dataset
    f))
 
+;; Recommendation: only call this within a temporary dataset.
+;; If calling within a fixed dataset, running the same test concurrently
+;; may fail due to exclusive dataset locking on snapshot deletion.
+;;
```

Comment:
IWBNI we could enforce this some way, but I guess all those set-up and tear-down calls also have to be synchronous.